core/src/main/java/org/apache/iceberg/ContentFileParser.java

```diff
@@ -48,6 +48,97 @@ class ContentFileParser {
 
   private ContentFileParser() {}
 
+  public static void unboundContentFileToJson(
+      ContentFile<?> contentFile, PartitionSpec spec, JsonGenerator generator) throws IOException {
+    Preconditions.checkArgument(contentFile != null, "Invalid content file: null");
+    Preconditions.checkArgument(spec != null, "Invalid partition spec: null");
+    Preconditions.checkArgument(generator != null, "Invalid JSON generator: null");
```

Comment:
We generally try to avoid exposing `JsonGenerator` or other Jackson classes in public methods. Can this be package-private? See [`SnapshotParser` for an example](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/SnapshotParser.java#L55) of how other parsers handle visibility.